Ccp 499 frontend solution to replace price

aaron-martin opened the pull request:
@sznowicki 

Please let me know if you don't have time to look at this pull request.

Thanks
Aaron

Commit:
CCP-396: set adjustProductsForMapPricing to return an empty price object rather than null when an item is a map product

Commit:
CCP-461: Added frontend components including portals for the price and strike out price to respond to MAP price settings.

Commit:
CCP-461: corrected formatting issue

Commit:
CCP-461: more formatting corrections

Commit:
CCP-499: created frontend portals to hide prices of map products

Commit:
CCP-461: reorganized files to eliminate duplcated code

Commit:
CCP-499: merged branch CCP-461 into this branch to avoid future conflicts.

Commit:
.gitignore update

Commit:
CCP-499: corrected issue with ios theme, and issues showing striked prices

Commit:
CCP-461: corrected extension name in package file, and reorganized components

Commit:
Merge branch 'CCP-461-prevent-variant-prices-of-map-products' of https://github.com/shopgate/ext-minimum-advertising-pricing into CCP-461-prevent-variant-prices-of-map-products

Commit:
CCP-499: corrected name in package file

Commit:
Merge remote-tracking branch 'origin/CCP-461-prevent-variant-prices-of-map-products' into CCP-499-frontend-solution-to-replace-price

# Conflicts:
#	frontend/pages/product/ProductDetailStrikePrice/components/PriceStriked/index.jsx

sznowicki reviewed (changes requested):
All looks good except one thing.

We should remove the adjustProductsForMapPricing backend extension.

We should not remove the price information when sending a product. Price is required for many features including some tracking scripts.

Also, most prominent problem is that price placeholder is never hidden because when price is empty frontend thinks it's not yet fetched.

Backend should only check is product is MAP product and add flags accordingly (as we have it now). It should not manipulate the price information, that's the frontend job since frontend knows where to show the prices and where it must be hidden.

Commit:
CCP-499: removed adjustMapProducts step from getProducts pipeline and  removed the supporting adjustMapProducts step file

aaron-martin commented:
@sznowicki 
I have removed the step that removes price data from the getProducts pipeline, and I removed the extension file that processed the work of that step.

Thanks
Aaron

sznowicki commented:
Cool. I'd remove the extension step from project extension as well ;)
